numpy/lib/stride_tricks.py

```diff
@@ -111,6 +115,225 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
+    dimensions of the array and extracts subsets of the array at all window
+    positions.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+    window_shape : int or tuple of int
+        Size of window over each axis that takes part in the sliding window.
+        If `axis` is not present, must have same length as the number of input
+        array dimensions. Single integers `i` are treated as if they were the
+        tuple `(i,)`.
+    axis : int or tuple of int, optional
+        Axis or axes along which the sliding window is applied.
+        By default, the sliding window is applied to all axes and
+        `window_shape[i]` will refer to axis `i` of `x`.
+        If `axis` is given as a `tuple of int`, `window_shape[i]` will refer to
+        the axis `axis[i]` of `x`.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+    subok : bool, optional
+        If True, sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+    writeable : bool, optional
+        When true, allow writing to the returned view. The default is false,
+        as this should be used with caution: the returned view contains the
+        same memory location multiple times, so writing to one location will
+        cause others to change.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window view of the array. The sliding window dimensions are
+        inserted at the end, and the original dimensions are trimmed as
+        required by the size of the sliding window.
+        That is, ``view.shape = x_shape_trimmed + window_shape``, where
+        ``x_shape_trimmed`` is ``x.shape`` with every entry reduced by one less
+        than the corresponding window size.
+
+    See Also
+    --------
+    lib.stride_tricks.as_strided: A lower-level and less safe routine for
+        creating arbitrary views from custom shape and strides.
+    broadcast_to: broadcast an array to a given shape.
+
+    Notes
+    -----
+    For some cases there may be more efficient approaches to calculate
+    transformations across multi-dimensional arrays, for instance
+    `scipy.signal.fftconvolve`, where combining the iterating step with the
+    calculation itself while storing partial results can result in significant
+    speedups.
+
+    As a rough estimate, a sliding window approach with an input size of `N`
+    and a window size of `W` will scale as `O(N*W)` where frequently a special
+    algorithm can achieve `O(N)`. That means that the sliding window variant
+    for a window size of 100 can be a 100 times slower than a more specialized
+    version.
+    For many common problems these better, specialized solutions already exist
+    in libraries such as
+    `scipy.ndimage <https://docs.scipy.org/doc/scipy/reference/ndimage.html>`_
+    or `bottleneck <https://github.com/pydata/bottleneck>`_.
```

Comment:
I like `window_shape`. I am OK with the current text. Personally, I would prefer it lead slightly stronger.  Not sure, maybe, leading with.

For many applications using a sliding view can be convenient but potentially very slow. Often specialized solutions solution exists, for example:

* `scipy.signal.fftconvolve`
* Filtering functions in `scipy.ndimage`
* Moving window functions provided by `bottleneck <https://bottleneck.readthedocs.io/en/latest/intro.html>`_

...

Reply:
But, at some point, we should probably call it a day and leave it to a future PR.